main/src/main/scala/sbt/EvaluateTask.scala

```diff
@@ -432,6 +435,7 @@ object EvaluateTask {
         shutdown()
       }
     }
+    currentlyRunningEngine.set((state, runningEngine))
```

Comment:
This should be unset after run() so it doesn't hold on to the reference.

Reply:
sounds good